Add support for device discovery and additional devices

reznet opened the pull request:
## What?

Implements prometheus http service discovery to discover supported devices so the static_configs list does not need to be used for maintained.  Also adds support for the outdoor meter and Hub 2.

## Why?

In Graphana, I could find no good way to display the device names - it only displayed device ids.  In addition, it didn't display any data for my outdoor meters or Hub2 devices.

## How?

A new endpoint `discover` is implemented which queries switchbot for all devices.  This list is filtered to those that are supported by the exporter (all devices that have temperature and humidity, as well as the mini plug) and returned to prometheus along with the device type and name.

For the new device types, since the exporter already supported temperature and humidity, the additional devices were added to the code that processes device status.  When a request comes in for metrics for an unsupported device type, a message is logged.

README is updated with additional example config using service discovery.

## Testing?

Tested with my own Meter, Hub2, and Outdoor meters.

nasa9084 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+# ignore vscode settings
+.vscode/
```
Can you end this file with a newline?

reznet replied:
fixed

reznet replied:
well, I think I fixed it.  On my Mac, it's 3 lines, but online it appears to be only two.  the PR includes the commit which adds the final newline.  let me know if it still doesn't look like it ends in a newline.

nasa9084 commented on main.go:
```diff
@@ -86,6 +93,50 @@ func run() error {
+		}
+		log.Printf("discovered device count: %d", len(devices))
+
+		supportedDeviceTypes := make(map[switchbot.PhysicalDeviceType]struct{})
+		supportedDeviceTypes[switchbot.Hub2] = struct{}{}
+		supportedDeviceTypes[switchbot.Humidifier] = struct{}{}
+		supportedDeviceTypes[switchbot.Meter] = struct{}{}
+		supportedDeviceTypes[switchbot.MeterPlus] = struct{}{}
+		supportedDeviceTypes[switchbot.PlugMiniJP] = struct{}{}
+		supportedDeviceTypes[switchbot.WoIOSensor] = struct{}{}
```
You can just do with a composite literal:

```suggestion
		supportedDeviceTypes := map[switchbot.PhysicalDeviceType]struct{}{
			switchbot.Hub2: struct{}{},
			switchbot.Humidifier: struct{}{},
			switchbot.Meter: struct{}{},
			switchbot.MeterPlus: struct{}{},
			switchbot.PlugMiniJP: struct{}{},
			switchbot.WoIOSensor: struct{}{},
		}
```

reznet replied:
thanks, I'm brand new to go.  I applied an additional simplification suggested by the golang plugin in vs code.

nasa9084 reviewed (changes requested):
Thank you for your great work, I left two comments:

reznet commented:
Hi @nasa9084, I'm a bit new to PR's in GitHub.  I believe I've addressed your comments and this PR is ready for re-review, but please let me know if I skipped a step or didn't quite follow the GitHub PR process correctly.  Thanks!

nasa9084 reviewed (approved):
@reznet Sorry for being late (I was too busy for a family matter for a month...) 🙇 
It looks great job, thank you for your contributing even though you are new to GitHub PR (trying new things should always be frustrating (but fun?))
